### ISSUE TITLE
Localization error for known locales during setup - the page shows in English

### ISSUE THREAD

**Issue by nemanjaglumac** [metabase/metabase#13063]
**Describe the bug**
Using the browser with some of the known/supported locales (fr, pt, de...) setup page is shown in English. It happens either immediately (v0.37.0-snapshot), or the page initially loads in the chosen language only to be displayed in English after page refresh (Docker v0.36.1).

**Logs**
See: Screenshots.

**To Reproduce**
### Prepare your browser
1. set the browsing language to French, for example
2. restart browser
3. open any page and go to browser console
4. type in `navigator.language` - it should give you `"fr"`

### v0.37.0-snapshot
1. Delete your database to make sure you start from scratch
2. Go to `/` (usually `localhost:3000`)
2a. it will redirect you to `/setup`
3. The setup page renders in English
4. open console and observe errors (see: **Screenshot 1**)
5. Click on "Let's get started"
6. In the list of preferred languages, French is pre-selected, but the whole page is rendered in English
7. Click on any other language in the list (for example: Italian)
8. The page is still displayed in English
9. The console displays new errors (see: **Screenshot 2**)

### Docker v0.36.1
1. if you are running in parallel with the local dev, choose port different than 3000 
2. Go to `/` (in my case `localhost:3100`)
2a. it will redirect you to `/setup`
3. The setup page renders in French, as expected
4. open console and observe the error seen in **Screenshot 3**
5. Refresh the page
6. It is now shown in English
6a. _No matter how many times you refresh the page it will still be displayed in English. But, close the browser and open `localhost:3100` again, the page will initially render in French, until you refresh it and the cycle repeats as described._
7. Weirdly, console doesn't show any errors
8. Click on "Let's get started"
9. In the list of preferred languages, French is pre-selected, but the whole page is rendered in English
10. Click on any other language (for example: Italian)
11. The page now renders correctly in the selected language

**Expected behavior**
Pages should immediately be translated to the language set in the browser, if that language is among the available/supported locales in Metabase `locales.clj`, and page reload shouldn't affect the language.

**Screenshot 1**
![image](https://user-images.githubusercontent.com/31325167/89311259-0e24d000-d676-11ea-9c35-20e15a24180a.png)

**Screenshot 2**
![image](https://user-images.githubusercontent.com/31325167/89316185-0ec06500-d67c-11ea-9b95-f128ab08e529.png)

**Screenshot 3**
![image](https://user-images.githubusercontent.com/31325167/89315246-ebe18100-d67a-11ea-8dfc-a93e94aeb387.png)

**Information about your Metabase Installation:**
| local dev              | Docker |
|---                         |---         |
| v0.37.0-snapshot|v0.36.1  |

- Your browser and the version: any browser
- Your operating system: OS X 10.15.6
- Your databases: N/a 
- Metabase version: see above
- Metabase hosting environment: see above
- Metabase internal database: H2 (default)

**Severity**
It really depends on the user and where they come from, how good is their English and whether they are willing to use Metabase in English instead of their own language.

**Additional context**
- discovered during writing of #13054
- probably related to #13045.

**Comment by nemanjaglumac** [comment]
@flamber - I've found 2 more localization related bugs - both are spotted on Docker latest image (v0.36.3). Is it ok if I keep them in this thread?
I'll attach the gif/video screenshots. If you need me to separate them or document them in more detail, just let me know.

1. During setup, Chinese HKG and Chinese Taiwanese are not working.
https://share.getcloudapp.com/5zuG5WGK (GIF)

2. Account Settings language not working after sign out:
- Start with English as your default language. 
- Update language in Account settings,
- sign out and sign in again - everything is in English except the word "table".
- Go to account settings again and you'll see that the chosen language is NOT English.
https://share.getcloudapp.com/kpubdEy6 (MP4 - for some reason, it doesn't open in browser - click to download it)

**Comment by flamber** [comment]
@nemanjaglumac Dang it. 
1. This problem probably belongs to this issue. I have a feeling that it's caused by country locales (`zh-HK`, instead of just `zh`), which is something that was also introduced in 0.36, but I'm not totally sure (related issue 8964, not linking).
2. That deserves a new issue, since it's not really related to this or setup. A workaround, after you signin, reload the browser, then it shows in the language of the account.

**Comment by nemanjaglumac** [comment]
I checked the state for `currentUser`, and the locale field shows `zh_TW`. I couldn't find any code that would change `zh-TW` to an underscore version when pushing to the store. o.O

`MetabaseSettings.get("available-locales")` shows kebab case versions (as it should).

**Comment by flamber** [comment]
There has been a lot of changes in 0.35, were underscores was changed to dashes - this caused some problems with specific settings, so I bet it's related to that.

**Comment by nemanjaglumac** [comment]
@flamber Found it!
`src/metabase/util/i18n/impl.clj:19` 

`normalized-locale-string` is creating underscore versions. So, it seems that backend uses underscore notation and frontend (along with browsers) is using hypen ("kebab") notation.

**Comment by gusaiani** [comment]
@nemanjaglumac, is this issue still relevant?

**Comment by nemanjaglumac** [comment]
>  is this issue still relevant?

It doesn't seem so 🎉

![image](https://user-images.githubusercontent.com/31325167/134515779-ee3e3855-7540-40fb-a5f5-7bc50d5894d7.png)
![image](https://user-images.githubusercontent.com/31325167/134515868-8a650761-4fdb-4729-9e68-b83d6c3dcff1.png)

**Comment by nemanjaglumac** [comment]
@rlotun @flamber - we can close this now. Tbh, not sure at what point was this fixed.
Do we need to investigate that further?

**Comment by flamber** [comment]
I'm 99% sure it was fixed in 0.39 with all the dash vs underscore changes. But I'm not going to test it, so closing.
